src/ORM/EagerLoader.php

```diff
@@ -631,7 +632,8 @@ public function associationsMap($table)
                     'canBeJoined' => $canBeJoined,
                     'entityClass' => $instance->target()->entityClass(),
                     'nestKey' => $canBeJoined ? $assoc : $meta->aliasPath(),
-                    'matching' => $forMatching !== null ? $forMatching : $matching
+                    'matching' => $forMatching !== null ? $forMatching : $matching,
+                    'targetProperty' => $meta->targetProperty()
```

Comment:
Not entirely related to this change, but why do we map the `EagerLoadable` instances into arrays? Could we just pass the objects around? Is that possibly an artifact of `EagerLoadable` being added late-ish in the development of the ORM?